### PR TITLE
KATI-120_MessageListFunction

### PR DESCRIPTION
**Pull request by toannguyen71195**

### CODE REVIEW

File: app/src/main/java/com/kms/cura/model/request/MessageModelResponse.java [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+
+/**
+ * Created by toanbnguyen on 7/25/2016.
+ */
+public class MessageModelResponse implements EntityModelResponse {
+
+    List<MessageEntity> messageEntities;
+    boolean gotResponse;
+    boolean responseError = false;
+    String error;
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/model/request/MessageModelResponse.java]: Make all the fields private

File: app/src/main/java/com/kms/cura/model/request/MessageModelResponse.java [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+                        doctorUserEntity = gson.fromJson(doctor.toString(), DoctorUserEntity.getDoctorEntityType());
+                        entity.setSender(doctorUserEntity);
+                        entity.setReceiver(patientUserEntity);
+                    } else {
+                        JSONObject patient = object.getJSONObject(MessageEntity.SENDER);
+                        patientUserEntity = gson.fromJson(patient.toString(), PatientUserEntity.getPatientUserType());
+                        JSONObject doctor = object.getJSONObject(MessageEntity.RECEIVER);
+                        doctorUserEntity = gson.fromJson(doctor.toString(), DoctorUserEntity.getDoctorEntityType());
+                        entity.setSender(patientUserEntity);
+                        entity.setReceiver(doctorUserEntity);
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/model/request/MessageModelResponse.java]: Bring 

```
patientUserEntity = gson.fromJson(patient.toString(), PatientUserEntity.getPatientUserType());
 doctorUserEntity = gson.fromJson(doctor.toString(), DoctorUserEntity.getDoctorEntityType());
entity.setSender(patientUserEntity);
entity.setReceiver(doctorUserEntity);
```

outside of if block to remove duplicated code

File: app/src/main/java/com/kms/cura/model/request/MessageUpdateModelResponse.java [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+import org.json.JSONObject;
+
+/**
+ * Created by toanbnguyen on 7/25/2016.
+ */
+public class MessageUpdateModelResponse implements EntityModelResponse {
+
+    boolean gotResponse;
+    boolean responseError = false;
+    String error;
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/model/request/MessageUpdateModelResponse.java]: Make fields private

File: app/src/main/java/com/kms/cura/view/fragment/MessageListFragment.java [hunk 4]
```diff
@@ -75,71 +78,49 @@ private void modifyToolbar(String title) {
...
+        new AsyncTask<Object, Void, Void>() {
+            private Exception exception = null;
+            ProgressDialog pDialog;
+            List<MessageEntity> messageEntities;
+
+            @Override
+            protected void onPreExecute() {
+                if (showDialog) {
+                    pDialog = new ProgressDialog(getActivity());
+                    pDialog.setMessage("Loading...");
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/fragment/MessageListFragment.java]: Put "Loading..." in resource file

File: app/src/main/java/com/kms/cura/view/fragment/MessageListFragment.java [hunk 5]
```diff
@@ -75,71 +78,49 @@ private void modifyToolbar(String title) {
...
+                return null;
+            }
 
+            @Override
+            protected void onPostExecute(Void aVoid) {
+                if (showDialog) {
+                    pDialog.dismiss();
+                } else {
+                    refreshLayout.setRefreshing(false);
+                    Toast.makeText(getActivity(), "Refreshed", Toast.LENGTH_SHORT).show();
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/fragment/MessageListFragment.java]: Put "Refreshed" in resource file

File: app/src/main/java/com/kms/cura/view/fragment/MessageListFragment.java [hunk 6]
```diff
@@ -237,4 +223,51 @@ public void onBackPressed() {
...
+    private void deleteThreads(final MessageThreadEntity entity) {
+        new AsyncTask<Object, Void, Void>() {
+            private Exception exception = null;
+            ProgressDialog pDialog;
+            List<MessageEntity> messageEntities;
+
+            @Override
+            protected void onPreExecute() {
+                pDialog = new ProgressDialog(getActivity());
+                pDialog.setMessage("Loading...");
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/fragment/MessageListFragment.java]: "Loading..." should be in resource file too

File: app/src/main/java/com/kms/cura/view/fragment/MessageListFragment.java [hunk 7]
```diff
@@ -237,4 +223,51 @@ public void onBackPressed() {
...
+
+            @Override
+            protected void onPostExecute(Void aVoid) {
+                pDialog.dismiss();
+                if (exception != null) {
+                    ErrorController.showDialog(getActivity(), "Error : " + exception.getMessage());
+                } else {
+                    adapter.setData(messageThreadEntities);
+                    adapter.notifyDataSetChanged();
+                    Toast.makeText(getActivity(), "Deleted", Toast.LENGTH_SHORT).show();
```
Comment (hieumai) [inline app/src/main/java/com/kms/cura/view/fragment/MessageListFragment.java]: "Deleted" should be in resource file too

File: com.kms.cura_server/src/com/kms/cura_server/MessageAPI.java [hunk 8]
```diff
@@ -56,6 +76,24 @@ public String deleteMessage(String jsonData) {
...
+            entity.setSender(doctorUserEntity);
+            entity.setReceiver(patientUserEntity);
+        } else {
+            JSONObject patient = object.getJSONObject(MessageEntity.SENDER);
+            patientUserEntity = JsonToEntityConverter.convertJsonStringToEntity(patient.toString(), PatientUserEntity.getPatientUserType());
+            JSONObject doctor = object.getJSONObject(MessageEntity.RECEIVER);
+            doctorUserEntity = JsonToEntityConverter.convertJsonStringToEntity(doctor.toString(), DoctorUserEntity.getDoctorEntityType());
+            entity.setSender(patientUserEntity);
+            entity.setReceiver(doctorUserEntity);
+        }
```
Comment (hieumai) [inline com.kms.cura_server/src/com/kms/cura_server/MessageAPI.java:85]: re-factor this code block the same as mentioned above

**Comment by hieumai** [comment]
OK from me. @huuduy216  and @vtnlinh95  please start verifying on this

**Comment by vtnlinh95** [comment]
@toannguyen71195  sometimes after deleting one thread , that thread does not disappear

**Comment by hieumai** [comment]
@toannguyen71195 please rebase this

**Pull request merged**